clloader/loader.py

```diff
@@ -114,46 +51,88 @@ def __init__(
             raise NotImplementedError(f"Evaluate mode {evaluate_on} is not supported.")
         self.evaluate_on = evaluate_on
 
-        self._setup(class_order)
+        self._setup(increment, initial_increment, class_order)
 
-        self.increments = self._define_increments(increment, initial_increment)
+    def _setup(self,increment: Union[List[int], int],
+               initial_increment: int,
+               class_order: Union[None, List[int]] = None) -> None:
 
-    def _define_increments(self, increment: Union[List[int], int],
-                           initial_increment: int) -> List[int]:
-        if isinstance(increment, list):
-            return increment
-        increments = []
-        if initial_increment:
-            increments.append(initial_increment)
-
-        nb_tasks = (self.nb_classes - initial_increment) / increment
-        if not nb_tasks.is_integer():
-            raise Exception(
-                "The tasks won't have an equal number of classes"
-                f" with {len(self.class_order)} and increment {increment}"
-            )
-        increments.extend([increment for _ in range(int(nb_tasks))])
-
-        return increments
-
-    def _setup(self, class_order: Union[None, List[int]] = None) -> None:
         (train_x, train_y), (test_x, test_y) = self.cl_dataset.init()
         unique_classes = np.unique(train_y)
 
         self.class_order = class_order or self.cl_dataset.class_order or list(
             range(len(unique_classes))
         )
+
+
         if len(np.unique(self.class_order)) != len(self.class_order):
             raise ValueError(f"Invalid class order, duplicates found: {self.class_order}.")
 
         mapper = np.vectorize(lambda x: self.class_order.index(x))
         train_y = mapper(train_y)
         test_y = mapper(test_y)
 
-        self.train_data = (train_x, train_y)
-        self.test_data = (test_x, test_y)
+        # Dataset without task label (necessary for self._define_increments())
+        self.train_data = (train_x, train_y)  # (data, class label, task label)
+        self.test_data = (test_x, test_y)  # (data, class label, task label)
         self.class_order = np.array(self.class_order)
 
+        # Increments setup
+        self.increments = self._define_increments(increment, initial_increment)
+
+        # compute task label
+        train_t = self._set_task_labels(train_y, self.increments)
+        test_t = self._set_task_labels(test_y, self.increments)
+
+        # Dataset with task label
+        self.train_data = (train_x, train_y, train_t)  # (data, class label, task label)
+        self.test_data = (test_x, test_y, test_t)  # (data, class label, task label)
+
+
+    def _set_task_labels(self, y: np.ndarray, increments: List[int])-> np.ndarray:
```

Comment:
It's more a `_get_task_labels`, you don't set anything here.

Reply:
I use this function to "create" the task label vector in the setup, that is why I put set. But I wasn't sure about the function name ^^